components/google_tag_manager/actions/create-tag/create-tag.mjs

```diff
@@ -0,0 +1,116 @@
+import app from "../../google_tag_manager.app.mjs";
+
+export default {
+  name: "Create Tag",
+  version: "0.0.1",
+  key: "google_tag_manager-create-tag",
+  description: "Create a tag in a workspace. [See documentation here](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/create)",
+  type: "action",
+  props: {
+    app,
+    accountId: {
+      optional: false,
+      propDefinition: [
+        app,
+        "accountId",
+      ],
+    },
+    containerId: {
+      optional: false,
+      propDefinition: [
+        app,
+        "containerId",
+        (c) => ({
+          accountId: c.accountId,
+        }),
+      ],
+    },
+    workspaceId: {
+      optional: false,
+      propDefinition: [
+        app,
+        "workspaceId",
+        (c) => ({
+          accountId: c.accountId,
+          containerId: c.containerId,
+        }),
+      ],
+    },
+    name: {
+      propDefinition: [
+        app,
+        "name",
+      ],
+      optional: true,
+    },
+    type: {
+      propDefinition: [
+        app,
+        "type",
+      ],
+    },
+    liveOnly: {
+      propDefinition: [
+        app,
+        "liveOnly",
+      ],
+    },
+    notes: {
+      propDefinition: [
+        app,
+        "notes",
+      ],
+    },
+    parameter: {
+      propDefinition: [
+        app,
+        "parameter",
+      ],
+    },
+    consentSettings: {
+      propDefinition: [
+        app,
+        "consentSettings",
+      ],
+    },
+    monitoringMetadata: {
+      propDefinition: [
+        app,
+        "monitoringMetadata",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const parameter = typeof this.parameter === "string"
```

Comment:
you could move this logic to a method, and return the `JSON.parse` or itself accordingly, for better reusability.